tools/clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -1047,8 +1047,15 @@ bool CGDebugInfo::TryCollectHLSLRecordElements(const RecordType *Ty,
     unsigned VecSize = hlsl::GetHLSLVecSize(QualTy);
     unsigned ElemSizeInBits = CGM.getContext().getTypeSize(ElemQualTy);
     unsigned CurrentAlignedOffset = 0;
+    std::string FieldNameBuf;
     for (unsigned ElemIdx = 0; ElemIdx < VecSize; ++ElemIdx) {
-      StringRef FieldName = StringRef(&"xyzw"[ElemIdx], 1);
+      StringRef FieldName;
+      if (VecSize <= 4) {
+        FieldName = StringRef(&"xyzw"[ElemIdx], 1);
+      } else {
+        FieldNameBuf = "c" + std::to_string(ElemIdx);
+        FieldName = FieldNameBuf;
```

Comment:
This is at least one, but probably two calls to `new`.

There are a couple different approaches we could take to reduce this. One would be a `Twine`, the other would be to have a `raw_svector_ostream` print directly into `FieldNameBuf`.
That would be something like:
```suggestion
        FieldNameBuf.clear();
        raw_svector_ostream OS(FieldNameBuf);
        OS << "c" << ElemIdx;
        FieldName = OS.str();
```

Reply:
Using `raw_svector_ostream` now